pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -56,3 +56,81 @@ func (out *OutboundTrafficPolicy) AddRoute(httpRouteMatch HTTPRouteMatch, weight
 	}
 	return nil
 }
+
+// MergeInboundPolicies merges latest InboundTrafficPolicies into original InboundTrafficPolicies
+func MergeInboundPolicies(original []*InboundTrafficPolicy, latest ...*InboundTrafficPolicy) []*InboundTrafficPolicy {
+	for _, l := range latest {
+		foundHostnames := false
+		for _, or := range original {
+			if reflect.DeepEqual(or.Hostnames, l.Hostnames) {
```

Comment:
Is there a scenario where the hostnames would overlap but not be equal? Is the list of hostnames guaranteed to always be a complete set? If so, how?